src/main/scala/hellotest/WordCloud.scala

```diff
@@ -5,8 +5,39 @@ class WordCloud(c: Int, l: Int, w: Int) {
   private val minLength = l
   private val windowSize = w
 
-  def process(input: Iterator[String], output: OutputObserver): Unit = 
-  {
-    // DO: create word cloud from given iterator and call observer so it can display result to user
+  def process(input: Iterator[String], output: OutputObserver): Unit = {
+    var window = scala.collection.mutable.Queue[String]() // Sliding window for the last 'windowSize' words
+    var wordFrequency = scala.collection.mutable.Map[String, Int]() // To track word frequencies
+
+    input.foreach { word =>
+      // Step 1: Filter out words that are shorter than 'minLength'
+      if (word.length >= minLength) {
+        // Step 2: Add the word to the sliding window
+        window.enqueue(word)
+
+        // Step 3: Update the frequency map
+        wordFrequency(word) = wordFrequency.getOrElse(word, 0) + 1
+
+        // Step 4: If the window exceeds 'windowSize', remove the oldest word
+        if (window.size > windowSize) {
+          val oldestWord = window.dequeue() // Dequeue removes the oldest word
+
+          // Decrease frequency of the removed word
+          wordFrequency(oldestWord) -= 1
+          if (wordFrequency(oldestWord) == 0) {
+            wordFrequency.remove(oldestWord) // Remove the word if its count reaches zero
+          }
+        }
+
+        // Step 5: Sort by frequency and pick the top 'cloudSize' words
+        val topWordsMap: scala.collection.mutable.LinkedHashMap[String, Int] =
+          scala.collection.mutable.LinkedHashMap[String, Int](
+            wordFrequency.toSeq.sortBy(-_._2).take(cloudSize): _*
+          )
+
+        // Pass the topWordsMap to the output observer
+        output.output(topWordsMap.clone().asInstanceOf[output.Result])
```

Comment:
This can be simplified:
```
val mapIterator = wordFrequency.toSeq.sortWith((x,y) => x._2 > y._2 || (x._2 == y._2 && x._1 > y._1)).iterator

output.output(mapIterator.asInstanceOf[output.Result])
```
I'm not sure if there's a prettier way to do the sorting since we need to sort first by reverse alphabetical order and then reverse numerical order.

The reason that it's not passing the unit tests is because it's calling the output observer before 5 words have been recorded in `window`, so you need a check like this:
```
if(window.size == windowSize)
    output.output(mapIterator.asInstanceOf[output.Result])
```